### ISSUE TITLE
Gap in Snowflake `CREATE HYBRID TABLE` Parsing

### ISSUE THREAD

**Issue by JStantonC6** [sqlfluff/sqlfluff#6696]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff fails with a parse error on basic `CREATE HYBRID TABLE(...)` statements when the `snowflake` dialect is active.

### Expected Behaviour

sqlfluff should be able to parse this statement

### Observed Behaviour

sqlfluff fails with a parsing error for a valid Snowflake DDL statement: https://docs.snowflake.com/en/sql-reference/sql/create-hybrid-table

### How to reproduce

Place this into a file named `hybrid.sql`:

```sql
CREATE HYBRID TABLE foo (id NUMBER PRIMARY KEY);
```

Run `sqlfluff lint --dialect snowflake hybrid.sql` and see that it fails with a parse error:

```
== [hybrid.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | HYBRID TABLE foo (id NUMBER PRIMA...'
WARNING: Parsing errors found and dialect is set to 'snowflake'. Have you configured your dialect correctly?
```

### Dialect

snowflake

### Version

```console
$ python --version
Python 3.13.2

$ sqlfluff --version
sqlfluff, version 3.3.1
```

### Configuration

No `.sqlfluff` or `.sqlfluffignore` files were present in the folder or any parent folder where I was reproducing this issue

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
